python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -1669,6 +1669,60 @@ def get_materialization_count_by_partition(
 
         return materialization_count_by_partition
 
+    def get_latest_asset_materialization_planned_event_by_partition(
+        self, asset_key: AssetKey
+    ) -> Mapping[str, EventLogRecord]:
```

Comment:
Is there a reason why we need the full event log record? Generally fetching the full event log record is much more expensive than fetching just the asset key and partition key due to deserialization costs.

If we can display everything we need using just the asset key, partition, and storage ID from the row, I think we should try to do that